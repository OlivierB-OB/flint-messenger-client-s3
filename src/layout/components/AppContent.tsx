import Box from '@material-ui/core/Box';
import React from 'react';
import { Switch, Route } from 'react-router-dom';
import { LoginScreen } from '../../login/components/LoginScreen';
import { Chat } from '../../conversations/components/Chat';
import { MyProfile } from '../../profileForm/components/MyProfile';
import { Call } from '../../call/components/Call';

export function AppContent() {
  return (
    <Box style={{ height: '90vh' }}>
      <Switch>
        <Route path="/login" component={LoginScreen} />
        <Route path="/profile" component={MyProfile} />
        <Route path="/conversation/:conversationId" component={Chat} />
        <Route path="/call" component={Call} />
      </Switch>
    </Box>
  );
}
