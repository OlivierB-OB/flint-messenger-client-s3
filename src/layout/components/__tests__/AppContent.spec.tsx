import React from 'react';
import { shallow } from 'enzyme';
import { AppContent } from '../AppContent';

describe('AppContent', () => {
  it('should take care of the routing', async () => {
    const component = shallow(<AppContent />);
    expect(component).toMatchSnapshot();
  });
});
