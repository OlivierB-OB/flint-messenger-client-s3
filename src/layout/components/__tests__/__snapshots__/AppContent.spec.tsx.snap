// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContent should take care of the routing 1`] = `
<Styled(MuiBox)
  style={
    Object {
      "height": "90vh",
    }
  }
>
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(LoginScreenDisplay)",
          "type": [Function],
        }
      }
      path="/login"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(ProfileForm)",
          "type": [Function],
        }
      }
      path="/profile"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(ChatDisplay)",
          "type": [Function],
        }
      }
      path="/conversation/:conversationId"
    />
  </Switch>
</Styled(MuiBox)>
`;
